lib/src/posthog_widget_widget.dart

```diff
@@ -0,0 +1,116 @@
+import 'dart:async';
+import 'dart:ui' as ui;
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/material.dart';
+import 'package:posthog_flutter/posthog_flutter.dart';
+import 'package:posthog_flutter/src/replay/mask/posthog_mask_controller.dart';
+import 'package:posthog_flutter/src/util/logging.dart';
+
+import 'replay/change_detector.dart';
+import 'replay/native_communicator.dart';
+import 'replay/screenshot/screenshot_capturer.dart';
+
+class PostHogWidget extends StatefulWidget {
+  final Widget child;
+
+  const PostHogWidget({super.key, required this.child});
+
+  @override
+  PostHogWidgetState createState() => PostHogWidgetState();
+}
+
+class PostHogWidgetState extends State<PostHogWidget> {
+  ChangeDetector? _changeDetector;
+  ScreenshotCapturer? _screenshotCapturer;
+  NativeCommunicator? _nativeCommunicator;
+
+  Timer? _debounceTimer;
+  Duration _debounceDuration = const Duration(milliseconds: 1000);
+
+  @override
+  void initState() {
+    super.initState();
+
+    final config = Posthog().config;
+    if (config == null || !config.sessionReplay) {
+      return;
+    }
+
+    _debounceDuration = config.sessionReplayConfig.debouncerDelay;
+
+    _screenshotCapturer = ScreenshotCapturer(config);
+    _nativeCommunicator = NativeCommunicator();
+
+    _changeDetector = ChangeDetector(_onChangeDetected);
+    _changeDetector?.start();
+  }
+
+  // This works as onRootViewsChangedListeners
+  void _onChangeDetected() {
+    _debounceTimer?.cancel();
+
+    _debounceTimer = Timer(_debounceDuration, () {
```

Comment:
this is intentional for now since Flutter is single-threaded, better to lose snapshots than a negative performance hit, will change it if we see there's no issue for now, Android also does debounce and its fine for now since its not a common behavior to "click/click/click" in a very short time frame